src/modal/internal.tsx

```diff
@@ -96,12 +98,35 @@ function PortaledModal({
 
   // enable / disable body scroll
   useEffect(() => {
+    /**
+     * Resets performanceMetricLogged
+     * PerformanceMetricLogged false and componentLoadingCount 0,
+     * indicates that there were no components loading in the modal and it was loaded instantly.
+     * In that case emit 0 as timeToContentReadyInModal
+     */
+    const resetModalPerformanceData = () => {
+      if (loadStartTime.current !== 0 && componentLoadingCount.current === 0 && !performanceMetricLogged.current) {
+        PerformanceMetrics.modalPerformanceData({
+          timeToContentReadyInModal: 0,
+          instanceIdentifier: instanceUniqueId,
+        });
+      }
+      loadStartTime.current = 0;
+      performanceMetricLogged.current = false;
+    };
     if (visible) {
       disableBodyScrolling();
+      loadStartTime.current = performance.now();
     } else {
       enableBodyScrolling();
+      resetModalPerformanceData();
     }
-  }, [visible]);
+    return () => {
+      if (!visible) {
+        resetModalPerformanceData();
+      }
+    };
```

Comment:
I think I'm not quite following why we're running this here. This function runs when the Modal component either leaves the invisible state, or when it unmounts. Couldn't we instead just reset the data when the Modal _opens_, i.e. when the effect runs and `visible` is true? That way the (visible) modal will always start  from a clean state

Reply:
Will add a comment and a timer to emit the metric